dds/src/implementation/dds_impl/user_defined_subscriber.rs

```diff
@@ -176,28 +168,7 @@ impl DdsShared<UserDefinedSubscriber> {
     }
 
     pub fn data_reader_list(&self) -> DdsIterator<'_, UserDefinedDataReader> {

```

Comment:
I wonder: since the subscriber should represent a data reader list + extras. If the subscriber should not be the iterator itself, rather then an iterator factory.

Reply:
The subscriber itself can't be an iterator because it needs to be somehow consumed by the usage. It could maybe be an IntoIterator but we tried that previously and found it doesn't read so nice.